simulator/ssh-transfer.go

```diff
@@ -0,0 +1,158 @@
+package simulator
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"net"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	simssh "github.com/alphasoc/flightsim/simulator/ssh"
+
+	bytesize "github.com/inhies/go-bytesize"
+)
+
+const defaultSendSize = 100 * bytesize.MB
+
+// TODO where will the server side live?
+var defaultTargetHosts = []string{"127.0.0.1:22", "127.0.0.1:9999"}
+
+// SSHTransfer defines this simulation.
+type SSHTransfer struct {
+	src      net.IP // Connect from this IP.
+	sendSize bytesize.ByteSize
+}
+
+// NewSSHTransfer creates a new SSH/SFTP simulator.
+func NewSSHTransfer() *SSHTransfer {
+	return &SSHTransfer{}
+}
+
+// Init sets the source IP for this simulation.
+func (s *SSHTransfer) Init(src net.IP) error {
+	s.src = src
+	return nil
+}
+
+// writeRandom writes sendSize bytes of 'random' data to the server.  An error is returned.
+func writeRandom(c *simssh.Client, handle string, sendSize bytesize.ByteSize) error {
+	// 8K writes.
+	const buffSize uint64 = 8192
+	toSendStr := sendSize.Format("%.0f", "B", false)
+	toSend, err := strconv.ParseUint(toSendStr[:len(toSendStr)-1], 10, 64)
+	if err != nil {
+		return fmt.Errorf("failed write: %v", err)
+	}
+	// Seed once.
+	rand.Seed(time.Now().UnixNano())
+	var i uint64
+	var bytes []byte
+	var payloadSize uint64
+	for i = 0; toSend > 0; i++ {
+		if toSend >= buffSize {
+			payloadSize = buffSize
+		} else {
+			payloadSize = toSend
+		}
+		bytes = make([]byte, payloadSize)
```

Comment:
This is fine, but it would be nicer if you create a buffer before the loop and reuse it.

Reply:
You got it.